<ui version="4.0" >
 <comment> * OpenRPT report writer and rendering engine
 * Copyright (C) 2001-2012 by OpenMFG, LLC
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 * Please contact info@openmfg.com with any questions on this license.</comment>
 <class>GraphWindow</class>
 <widget class="QMainWindow" name="GraphWindow" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>677</width>
    <height>763</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>Graph Window</string>
  </property>
  <widget class="QWidget" name="widget" >
   <layout class="QVBoxLayout" >
    <property name="spacing" >
     <number>6</number>
    </property>
    <property name="margin" >
     <number>11</number>
    </property>
    <item>
     <layout class="QHBoxLayout" >
      <property name="spacing" >
       <number>6</number>
      </property>
      <property name="margin" >
       <number>0</number>
      </property>
      <item>
       <widget class="QFrame" name="_graphFrame" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Expanding" hsizetype="Expanding" >
          <horstretch>1</horstretch>
          <verstretch>1</verstretch>
         </sizepolicy>
        </property>
        <property name="frameShape" >
         <enum>QFrame::Box</enum>
        </property>
        <property name="frameShadow" >
         <enum>QFrame::Plain</enum>
        </property>
        <property name="lineWidth" >
         <number>2</number>
        </property>
        <layout class="QHBoxLayout" >
         <property name="spacing" >
          <number>6</number>
         </property>
         <property name="margin" >
          <number>2</number>
         </property>
         <item>
          <widget class="Graph" native="1" name="_graph" >
           <property name="sizePolicy" >
            <sizepolicy vsizetype="Minimum" hsizetype="Minimum" >
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize" >
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" >
        <property name="spacing" >
         <number>6</number>
        </property>
        <property name="margin" >
         <number>0</number>
        </property>
        <item>
         <widget class="QGroupBox" name="_gbLabel" >
          <property name="title" >
           <string>Labels</string>
          </property>
          <layout class="QGridLayout" >
           <property name="margin" >
            <number>11</number>
           </property>
           <property name="spacing" >
            <number>6</number>
           </property>
           <item row="1" column="0" >
            <widget class="QLabel" name="_dataLabel" >
             <property name="text" >
              <string>Data:</string>
             </property>
             <property name="alignment" >
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="buddy" >
              <cstring>_dataEdit</cstring>
             </property>
            </widget>
           </item>
           <item row="0" column="0" >
            <widget class="QLabel" name="_titleLabel" >
             <property name="text" >
              <string>Title:</string>
             </property>
             <property name="alignment" >
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="buddy" >
              <cstring>_titleEdit</cstring>
             </property>
            </widget>
           </item>
           <item row="2" column="0" >
            <widget class="QLabel" name="_valueLabel" >
             <property name="text" >
              <string>Value:</string>
             </property>
             <property name="alignment" >
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="buddy" >
              <cstring>_valueEdit</cstring>
             </property>
            </widget>
           </item>
           <item row="0" column="1" >
            <widget class="QLineEdit" name="_titleEdit" >
             <property name="sizePolicy" >
              <sizepolicy vsizetype="Fixed" hsizetype="Minimum" >
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
            </widget>
           </item>
           <item row="1" column="1" >
            <widget class="QLineEdit" name="_dataEdit" >
             <property name="sizePolicy" >
              <sizepolicy vsizetype="Fixed" hsizetype="Minimum" >
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
            </widget>
           </item>
           <item row="2" column="1" >
            <widget class="QLineEdit" name="_valueEdit" >
             <property name="sizePolicy" >
              <sizepolicy vsizetype="Fixed" hsizetype="Minimum" >
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="_gbPadding" >
          <property name="title" >
           <string>Padding (in pixels)</string>
          </property>
          <layout class="QGridLayout" >
           <property name="margin" >
            <number>11</number>
           </property>
           <property name="spacing" >
            <number>6</number>
           </property>
           <item row="1" column="0" >
            <widget class="QLabel" name="_vPadLabel" >
             <property name="text" >
              <string>Vertical:</string>
             </property>
             <property name="alignment" >
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="buddy" >
              <cstring>_vPadSpinbox</cstring>
             </property>
            </widget>
           </item>
           <item row="0" column="0" >
            <widget class="QLabel" name="_hPadLabel" >
             <property name="text" >
              <string>Horizontal:</string>
             </property>
             <property name="alignment" >
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="buddy" >
              <cstring>_hPadSpinbox</cstring>
             </property>
            </widget>
           </item>
           <item row="0" column="1" >
            <widget class="QSpinBox" name="_hPadSpinbox" >
             <property name="sizePolicy" >
              <sizepolicy vsizetype="Fixed" hsizetype="Minimum" >
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="value" >
              <number>5</number>
             </property>
            </widget>
           </item>
           <item row="1" column="1" >
            <widget class="QSpinBox" name="_vPadSpinbox" >
             <property name="sizePolicy" >
              <sizepolicy vsizetype="Fixed" hsizetype="Minimum" >
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="value" >
              <number>5</number>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="_gbValueRange" >
          <property name="title" >
           <string>Value Range</string>
          </property>
          <layout class="QGridLayout" >
           <property name="margin" >
            <number>11</number>
           </property>
           <property name="spacing" >
            <number>6</number>
           </property>
           <item row="0" column="0" >
            <widget class="QLabel" name="_maxValueLabel" >
             <property name="text" >
              <string>Max:</string>
             </property>
             <property name="alignment" >
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="buddy" >
              <cstring>_maxValueSpinbox</cstring>
             </property>
            </widget>
           </item>
           <item row="1" column="0" >
            <widget class="QLabel" name="_minValueLabel" >
             <property name="text" >
              <string>Min:</string>
             </property>
             <property name="alignment" >
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="buddy" >
              <cstring>_minValueSpinbox</cstring>
             </property>
            </widget>
           </item>
           <item row="1" column="1" >
            <widget class="QSpinBox" name="_minValueSpinbox" >
             <property name="sizePolicy" >
              <sizepolicy vsizetype="Fixed" hsizetype="Minimum" >
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimum" >
              <number>-999</number>
             </property>
             <property name="maximum" >
              <number>0</number>
             </property>
            </widget>
           </item>
           <item row="0" column="1" >
            <widget class="QSpinBox" name="_maxValueSpinbox" >
             <property name="sizePolicy" >
              <sizepolicy vsizetype="Fixed" hsizetype="Minimum" >
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimum" >
              <number>1</number>
             </property>
             <property name="maximum" >
              <number>999999</number>
             </property>
             <property name="value" >
              <number>100</number>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="_btnClear" >
          <property name="text" >
           <string>Clear All</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="_styleGroup" >
      <property name="title" >
       <string>Graph Style</string>
      </property>
      <layout class="QHBoxLayout" >
       <property name="spacing" >
        <number>6</number>
       </property>
       <property name="margin" >
        <number>11</number>
       </property>
       <item>
        <widget class="QCheckBox" name="_cbBars" >
         <property name="text" >
          <string>Bars</string>
         </property>
         <property name="checked" >
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="_cbLines" >
         <property name="text" >
          <string>Lines</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="_cbPoints" >
         <property name="text" >
          <string>Points</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" >
      <property name="spacing" >
       <number>6</number>
      </property>
      <property name="margin" >
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="_numSetLabel" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Preferred" hsizetype="Expanding" >
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text" >
         <string>Number of Sets:</string>
        </property>
        <property name="alignment" >
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="buddy" >
         <cstring>_numSetSpinbox</cstring>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="_numSetSpinbox" >
        <property name="minimum" >
         <number>1</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="_numRefLabel" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Preferred" hsizetype="Expanding" >
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text" >
         <string>Number of References:</string>
        </property>
        <property name="alignment" >
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="buddy" >
         <cstring>_numRefSpinbox</cstring>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="_numRefSpinbox" >
        <property name="minimum" >
         <number>1</number>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" >
      <property name="spacing" >
       <number>6</number>
      </property>
      <property name="margin" >
       <number>0</number>
      </property>
      <item>
       <widget class="QTableWidget" name="_labelTable" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Expanding" hsizetype="Minimum" >
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize" >
         <size>
          <width>80</width>
          <height>0</height>
         </size>
        </property>
        <column>
         <property name="text" >
          <string>Label</string>
         </property>
        </column>
       </widget>
      </item>
      <item>
       <widget class="QTableWidget" name="_dataTable" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Expanding" hsizetype="Expanding" >
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="_gbSql" >
      <property name="enabled" >
       <bool>false</bool>
      </property>
      <property name="title" >
       <string>Populate with SQL</string>
      </property>
      <layout class="QHBoxLayout" >
       <property name="spacing" >
        <number>6</number>
       </property>
       <property name="margin" >
        <number>11</number>
       </property>
       <item>
        <widget class="QTextEdit" name="_editSql" />
       </item>
       <item>
        <layout class="QVBoxLayout" >
         <property name="spacing" >
          <number>6</number>
         </property>
         <property name="margin" >
          <number>0</number>
         </property>
         <item>
          <spacer>
           <property name="orientation" >
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeType" >
            <enum>QSizePolicy::Expanding</enum>
           </property>
           <property name="sizeHint" stdset="0" >
            <size>
             <width>20</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="_btnSql" >
           <property name="text" >
            <string>Execute</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11" />
 <customwidgets>
  <customwidget>
   <class>Graph</class>
   <extends>QWidget</extends>
   <header>graph.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>_hPadSpinbox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>_graph</receiver>
   <slot>setHPadding(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>599</x>
     <y>194</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_vPadSpinbox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>_graph</receiver>
   <slot>setVPadding(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>599</x>
     <y>223</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_titleEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>_graph</receiver>
   <slot>setTitle(QString)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>561</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_dataEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>_graph</receiver>
   <slot>setDataLabel(QString)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>561</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_valueEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>_graph</receiver>
   <slot>setValueLabel(QString)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>561</x>
     <y>122</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_maxValueSpinbox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>_graph</receiver>
   <slot>setMaxValue(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>599</x>
     <y>294</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_minValueSpinbox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>_graph</receiver>
   <slot>setMinValue(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>599</x>
     <y>323</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_cbBars</sender>
   <signal>toggled(bool)</signal>
   <receiver>_graph</receiver>
   <slot>setDrawBars(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>44</x>
     <y>430</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_cbLines</sender>
   <signal>toggled(bool)</signal>
   <receiver>_graph</receiver>
   <slot>setDrawLines(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>256</x>
     <y>430</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_cbPoints</sender>
   <signal>toggled(bool)</signal>
   <receiver>_graph</receiver>
   <slot>setDrawPoints(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>467</x>
     <y>430</y>
    </hint>
    <hint type="destinationlabel" >
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
